### DISCUSSION TITLE
Can I const outside of my component instead of memoizing?

### DISCUSSION THREAD

**Discussion by Obiwarn** [TanStack/table#3774] [Q&A]
Dear @tannerlinsley,

first of all, let me thank you for being one of the greatest open source contributors in the known universe.

Secondly, whenever I should memoize or useCallback() something in the Tanstack, can I just use a const variable (global to my file) outside of my functional component block (if I only need it once for all components of that type)? If no, why?

Thank you so much.

Example:
```
import { useTable } from 'react-table'
 
 const columns = 
     () => [
       {
         Header: 'Column 1',
         accessor: 'col1', // accessor is the "key" in the data
       },
       {
         Header: 'Column 2',
         accessor: 'col2',
       },
     ]

 function App() {
   const data = React.useMemo(
     () => [
       {
         col1: 'Hello',
         col2: 'World',
       },
       {
         col1: 'react-table',
         col2: 'rocks',
       },
       {
         col1: 'whatever',
         col2: 'you want',
       },
     ],
     []
   )
 
  // ...
```

**Comment by tannerlinsley** [comment]
I’d your columns are never changing from state or props, then yes, do it outside of your component.

Tanner Linsley
